maestro-client/src/main/java/maestro/MaestroTimer.kt

```diff
@@ -23,10 +23,13 @@ object MaestroTimer {
         return null
     }
 
-    fun retryUntilTrue(timeoutMs: Long, block: () -> Boolean): Boolean {
+    fun retryUntilTrue(timeoutMs: Long, delayMs: Long? = null, block: () -> Boolean): Boolean {
         val endTime = System.currentTimeMillis() + timeoutMs
         do {
             try {
+                delayMs?.let {
+                    sleep(Reason.BUFFER, delayMs)
+                }
```

Comment:
Is this needed to fix clearState? If not, then please add this in a separate PR.

One issue I have with this approach is that the user is waiting 3 seconds for no reason. If we really want to put a delay there (which I'm not convinced that we need that), it would be better placed after the block has been evaluated once.

Reply:
sounds good, the thing is without this delay it was hanging almost every second time for me and might happen to other users as well. also smth more robust like RxJava or Coroutines would be a better idea here.

I'll create a separate PR and issue, lets discuss there.